tests/middleware/test_sandbox.py

```
@@ -0,0 +1,464 @@
+    async def test_blocks_filesystem_tool_without_permission(
+        self, middleware, mock_handler, mock_request_requiring_filesystem
+    ):
+        """Filesystem tool blocked when only NETWORK granted."""
+        from src.sandboxes.impl.seatbelt import SeatbeltSandbox
+
+        config = SandboxConfig(
+            name="network-only",
+            type=SandboxType.SEATBELT,
+            permissions=[SandboxPermission.NETWORK],
+        )
+        sandbox = SeatbeltSandbox(config, Path("/tmp"))
```

Ruff (0.14.8)

Probable insecure usage of temporary file or directory: "/tmp"

(S108)